### DISCUSSION TITLE
🚀 Feature: Cloud functions keep alive

### DISCUSSION THREAD

**Discussion by s3ppo** [appwrite/appwrite#11509] [Ideas]
### 🔖 Feature description

Hi all,

and first of all i love the new functions and also their speed :)

i run into a small problem with the cold start of sync functions.
if a function wasnt used for a longer time, it shuts the container down and the next start needs up to 5 seconds to bring a result to the app (UI).

so i think it would be very cool to have a option in the settings of a function where i can define a keep alive checkbox. if i check this, the shutwdown of the container does not happen and it would bring me faster responses to my front end.

or maybe a setting where i can define for how long i want the container warmed up.. for not critically performance or background task i can define 1 minute, and for sync functions that brings responses to the front end, i can define 24h.

please let me know if this is possible? :)

### 🎤 Pitch

pls see description :)

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by PineappleIOnic** [comment]
Hey There 👋 
Thank you for the pitch, it's a very cool idea!
While we are considering this I would like to note we already have an environment variable that allows you to set the inactive threshold for functions: `_APP_FUNCTIONS_INACTIVE_THRESHOLD`. It isn't separate for sync and async functions though as internally they both act very similarly.

**Comment by s3ppo** [comment]
@PineappleIOnic 
thanks for your answer, yes this is true and also use it at this moment this way. but if i set the threshold to 24h theres still the first call of the day pretty slow, so this is a bad user experience in my mind ;)

also i think i dont need it for all my functions, just for special functions that brings data to my front end.. with threshold paramter its active for all my containers.. and i also its not available in the appwrite console, which is also a bad user experience ;)

**Comment by gewenyu99** [comment]
> @PineappleIOnic thanks for your answer, yes this is true and also use it at this moment this way. but if i set the threshold to 24h theres still the first call of the day pretty slow, so this is a bad user experience in my mind ;)
> 
> also i think i dont need it for all my functions, just for special functions that brings data to my front end.. with threshold paramter its active for all my containers.. and i also its not available in the appwrite console, which is also a bad user experience ;)

I think we can consider having this be a console setting, per container. CC: @PineappleIOnic

**Comment by Ruboka** [comment]
Is setting this per function now possible? Or is the docker environment variable still counting for all project functions? 
Also this feature customizing individual functions could be applied to available memory and cpus etc too. `_APP_FUNCTIONS_MEMORY ; _APP_FUNCTIONS_CPUS` More or less all variables from the env file.

**Comment by eldadfux** [comment]
@s3ppo what runtime are you using that takes 5 second to cold start?

**Comment by s3ppo** [comment]
> @s3ppo what runtime are you using that takes 5 second to cold start?

Was in an older release now its about 200ms on a cold start versus 2ms on a warm start
